### PR TITLE
GNM AWS GR Path Fix, Zero Divide Fix, and Extra Error Logging

### PR DESCRIPTION
**Pull request by Dala0**
Fixes problem with path by getting it from Vidispine, fixes a division error, and adds logging for a possible error. @fredex42

### CODE REVIEW

File: gnmawsgr/tasks.py [hunk 1]
```diff
@@ -129,11 +129,18 @@ def valid_fields(self):
 def download_callback(rq, current_progress,total):
     if rq.status != 'DOWNLOADING':
         rq.status = 'DOWNLOADING'
+
+    try:
+        percent = 100*float(current_progress)/float(total)
+
+    except ZeroDivisionError as e:
```
Comment (fredex42) [inline gnmawsgr/tasks.py]: you know what, i think we should actually run a differnet logger.info line here.  Something like, logger.warning("{itemid} Trying to download but last chunk size was zero".format(rq.item_id))

File: gnmawsgr/tasks.py [hunk 2]
```diff
@@ -259,7 +266,10 @@ def do_glacier_restore(request_id,itemid,path):
     while True:
         try:
             with open(filename,'wb') as fp:
-                key.get_file(fp, cb=partial(download_callback, rq), num_cb=40)
+                try:
+                    key.get_file(fp, cb=partial(download_callback, rq), num_cb=40)
+                except AttributeError:
+                    logger.error("No Key found in S3. Object values: {0}".format(rq.__dict__))
```
Comment (fredex42) [inline gnmawsgr/tasks.py]: think we need to return here as well.  Could you also log out to Sentry?

File: gnmawsgr/views.py [hunk 3]
```diff
@@ -64,9 +70,15 @@ def dispatch(self, request, *args, **kwargs):
...
+    from portal.vidispine.iitem import ItemHelper
 
     itemid = request.GET.get('id', '')
-    path = request.GET.get('path', '')
+
+    ith = ItemHelper()
+
+    res = ith.getItemMetadata(itemid)
+
+    path = metadataValueInGroup('ExternalArchiveRequest','gnm_external_archive_external_archive_path',res['item'])
```
Comment (fredex42) [inline gnmawsgr/views.py]: what is the difference between r() and re()? can we make them into the same method?

File: gnmawsgr/tasks.py (hunk 2 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: I was not sure of the syntax for try so I looked it up. According to this (https://docs.python.org/2.7/tutorial/errors.html) after the code in except is run the programme contines by default.

Comment (Dala0) [inline gnmawsgr/tasks.py]: I have attempted to add code to talk to Sentry. Please see the commit below.

File: gnmawsgr/views.py (hunk 3 above)
Comment (Dala0) [inline gnmawsgr/views.py]: I think I had a good reason for making another method at the time, such as I thought we might need some to work slightly differently. I will have a look at the code and see if I can safely remove re.

Comment (Dala0) [inline gnmawsgr/views.py]: re is doing various different things, such as setting a field to RETRY so I suggest we keep it.

File: gnmawsgr/tasks.py (hunk 1 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: I have attempted to build this. Please see the commit below.

File: gnmawsgr/views.py (hunk 3 above)
Comment (fredex42) [inline gnmawsgr/views.py]: ok. could you look at factoring the common code into a common function and just keeping the different functionality? DRY programming, ya know :)

File: gnmawsgr/tasks.py [hunk 4]
```diff
@@ -129,11 +129,19 @@ def valid_fields(self):
...
-        cur=current_progress,
-        tot=total,
-        pc=100*float(current_progress)/float(total)))
+
+    try:
+        percent = 100*float(current_progress)/float(total)
+
+    except ZeroDivisionError as e:
+        percent = 0.0
+        logger.warning("{0} Trying to download but last chuck size was zero".format(rq.item_id))
```
Comment (fredex42) [inline gnmawsgr/tasks.py]: typo - chunk not chuck :)

File: gnmawsgr/tasks.py [hunk 5]
```diff
@@ -129,11 +129,19 @@ def valid_fields(self):
...
+
+    #logger.info("{itemid} Download in progress: {cur}/{tot}, {pc:.2f}%".format(
```
Comment (fredex42) [inline gnmawsgr/tasks.py]: best not to comment this out, just return from the exception handler above and leave this in

Comment (Dala0) [inline gnmawsgr/tasks.py]: Will do. According to Python document returning is not needed.

File: gnmawsgr/tasks.py (hunk 4 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: Done.

File: gnmawsgr/tasks.py (hunk 5 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: I have removed the comments.

File: gnmawsgr/tasks.py (hunk 1 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: I have put in the return.

File: gnmawsgr/tasks.py (hunk 2 above)
Comment (Dala0) [inline gnmawsgr/tasks.py]: I have put a return in.

File: gnmawsgr/templates/n.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
+        "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+    <title></title>
+</head>
+<body>
+    <font size="+1">
+        Item restore not started because a restore job for that item is already underway.
```
Comment (fredex42) [inline gnmawsgr/templates/n.html]: please can we change the filenames to something a little more descriptive? thanks

**Pull request merged**